pkg/services/ngalert/state/persist.go

```diff
@@ -10,8 +10,8 @@ import (
 type InstanceStore interface {
 	FetchOrgIds(ctx context.Context) ([]int64, error)
 	ListAlertInstances(ctx context.Context, cmd *models.ListAlertInstancesQuery) error
-	SaveAlertInstance(ctx context.Context, cmd *models.SaveAlertInstanceCommand) error
-	DeleteAlertInstance(ctx context.Context, orgID int64, ruleUID, labelsHash string) error
+	SaveAlertInstances(ctx context.Context, cmd ...models.AlertInstance) error
```

Comment:
I do not think it is needed because I can't see the use-case for that data. The state manager persists the state just for one single reason - recover after restart.

Reply:
This is a generic store though for state persistence, not a specialized store for recover after restart. Is there no use case where we'd want to know the IDs of the inserted states, for example?

Reply:
* It's not hard to implement that when we come up with a use-case that requires the IDs. Building it up-front adds allocations that we aren't currently going to use.
* In this particular case, the keys must be known up-front, as they're not assigned by the database.
* The existing implementation didn't update anything about the passed `cmd`. Incidentally, this is another reason to avoid that pattern - whether or not you get some interesting data returned in the command is not obvious and you have to remember on a case-by-case basis. Returning values (even if they're the same values, mutated) makes the function signature a clearer explanation of what's going on.